src/v/pandaproxy/client/client.cc

```diff
@@ -148,25 +153,32 @@ ss::future<kafka::fetch_response::partition> client::fetch_partition(
           return make_fetch_request(tp, offset, max_bytes, timeout);
       };
 
-    return ss::do_with(
-      std::move(build_request),
-      std::move(tp),
-      [this](auto& build_request, model::topic_partition& tp) {
-          return retry_with_mitigation(
-                   shard_local_cfg().retries(),
-                   shard_local_cfg().retry_base_backoff(),
-                   [this, &tp, &build_request]() {
-                       return _brokers.find(tp)
-                         .then([&tp, &build_request](shared_broker_t&& b) {
-                             return b->dispatch(build_request(tp));
+    return ss::with_gate(
```

Comment:
this is a good example of where a commit adding the one liner instances of `with_gate` then a separate commit with the reformatting is very useful for code review.

Reply:
thanks! so much easier to read now